nerdlandbot/commands/pets.py

```diff
@@ -0,0 +1,309 @@
+            return await ctx.send(message)
+
+        if category.lower() not in guild_data.pets_categories:
+            message = translate("pet_category_nonexistant", lang)
+            return await ctx.send(message)
+
+        
+        await guild_data.add_pet(name, ctx.author.id, category.lower())
+        pet_id = guild_data.pets_last_id
+        if ctx.message.attachments:
+            await resize_and_save(ctx, ctx.message.attachments[0], pet_id)
+            msg = translate("pet_added_succes", lang).format(
+                name.capitalize(), pet_id)
+            await ctx.send(msg)
+        else:
+            embed_title = translate("pet_embed_title", lang)
+            question = translate("waiting_question",
+                                 lang).format(name.capitalize())
+            embed = discord.Embed(
+                title=embed_title,
+                description=question,
+            )
+            await ctx.send(embed=embed)
+            try:
+                reaction = await ctx.bot.wait_for(
+                    "message", timeout=INTERACT_TIMEOUT, check=check(ctx.author)
+                )
+
+                if reaction.content == "0":
+                    await guild_data.delete_pet(pet_id)
+                    abort = translate("command_abort", lang)
+                    embed = discord.Embed(
+                        title=embed_title,
+                        description=abort,
+                    )
+                    return await ctx.send(embed=embed)
+
+                if reaction.attachments:
+                    await resize_and_save(ctx, reaction.attachments[0], pet_id)
+                    msg = translate("pet_added_succes", lang).format(
+                        name.capitalize(), pet_id
+                    )
+                    await ctx.send(msg)
+                else:
+                    msg = translate("pet_reaction_error", lang)
+                    await ctx.send(msg)
+
+            except asyncio.TimeoutError:
+                await guild_data.delete_pet(pet_id)
+                timeout = translate("command_timeout", lang)
+                embed = discord.Embed(
+                    title=embed_title,
+                    description=timeout,
+                )
+                return await ctx.send(embed=embed)
+
+    @commands.command(
+        name="post_pet",
+        aliases=["pet", "random_pet"],
+        brief="post_pet_brief",
+        help="post_pet_help",
+        usage="post_pet_usage",
+    )
+    @commands.guild_only()
+    async def post_pet(self, ctx: commands.Context, name: typing.Optional[str] = None):
+        lang, guild_data = await asyncio.gather(
+            culture(ctx),
+            get_guild_data(ctx.message.guild.id),
+        )
+
+        pets = guild_data.pets
+
+        if name == None:
+            pet_id = random.choice(list(pets.keys()))
+            guild_path = Path.cwd() / PETS_DIR_NAME / str(ctx.guild.id)
+            filepath = guild_path / f"{pet_id}.jpg"
```

Comment:
I would create a method/function because you're copy pasting this a few times. You're also using this during saving.

```python
def get_pet_path(pet_id: str) -> Path:
```